include/difficulty.h

```diff
@@ -0,0 +1,13 @@
+#ifndef GUARD_DIFFICULTY_H
+#define GUARD_DIFFICULTY_H
+
+#include "constants/difficulty.h"
+
+u32 GetCurrentDifficultyLevel(void);
```

Comment:
With `enum DifficultyLevel` being a thing, I wonder if some of these untyped numbers should be `enum DifficultyLevel`?

Reply:
[`e78c983` (#5337)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5337/commits/e78c983e1cbcdf24a0c32b983ea58f79de9d8c6a)

Reply:
I think @mrgriffin meant the return types and arguments, specifically of

```
void SetCurrentDifficultyLevel(u32);
u32 GetBattlePartnerDifficultyLevel(u16);
u32 GetTrainerDifficultyLevel(u16);
```
->
```
void SetCurrentDifficultyLevel(enum DifficultyLevel);
enum DifficultyLevel GetBattlePartnerDifficultyLevel(u16);
enum DifficultyLevel GetTrainerDifficultyLevel(u16);
```
(Probably needs a few more changes in the definitions / usages as well, otherwise I would have used a change suggestion)

Reply:
[`d896070` (#5337)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5337/commits/d896070e29ec161f38a8530b844e14db57455cb2)